src/server/bitops_family.cc

```diff
@@ -343,8 +343,14 @@ std::string ElementAccess::Value() const {
 
 void ElementAccess::Commit(std::string_view new_value) const {
   if (shard_) {
-    element_iter_->second.SetString(new_value);
-    post_updater_.Run();
+    if (new_value.empty() && IsNewEntry()) {
```

Comment:
I think we should delete it even if its not a new entry
set x y
bitop not x key
get x
should also return nil

Reply:
I also added a test and confirmed the behaviour with Valkey. good catch